### PR TITLE
DEV: add test cases for email notifications about channel-wide mentions

### PR DESCRIPTION
**Pull request by AndrewPrigorshnev**
A follow-up to https://github.com/discourse/discourse/pull/20677.

### CODE REVIEW

File: plugins/chat/spec/mailers/user_notifications_spec.rb [hunk 1]
```diff
@@ -161,6 +161,74 @@ def refresh_auto_groups
...
+          # and we have to create it like here. In this case all the necessary
+          # db records for mentions and notifications will be created under the hood.
+          Chat::ChatMessageCreator.create(chat_channel: channel, user: sender, content: content)
+        end
+
+        it "returns email for @all mention by default" do
+          create_chat_message_with_mentions_and_notifications("Mentioning @all")
+          email = described_class.chat_summary(user, {})
+          expect(email.to).to be_present
+          expect(email.to).to contain_exactly(user.email)
```
Comment (jjaffeux) [inline plugins/chat/spec/mailers/user_notifications_spec.rb:178]: Same question than after

**Review by jjaffeux — APPROVED** [review]
Great thanks for not dropping the ball on this one :+1:

File: plugins/chat/spec/mailers/user_notifications_spec.rb [hunk 2]
```diff
@@ -161,6 +161,74 @@ def refresh_auto_groups
...
+        end
+
+        it "returns email for @here mention by default" do
+          user.update(last_seen_at: 1.second.ago)
+
+          create_chat_message_with_mentions_and_notifications("Mentioning @here")
+          email = described_class.chat_summary(user, {})
+
+          expect(email.to).to be_present
+          expect(email.to).to contain_exactly(user.email)
```
[the remark this answers is not in the document]
Comment (AndrewPrigorshnev) [reply plugins/chat/spec/mailers/user_notifications_spec.rb]: Oh, you're right indeed, the second line is enough.

File: plugins/chat/spec/mailers/user_notifications_spec.rb (hunk 1 above)
Comment (AndrewPrigorshnev) [reply plugins/chat/spec/mailers/user_notifications_spec.rb]: I've fixed this one too.

**Pull request merged**